### DISCUSSION TITLE
Issues with SPA Navigation Between app and page Directories in Next.js

### DISCUSSION THREAD

**Discussion by shunshimono** [vercel/next.js#68261] [Help]
Hello,

In my Next.js project, I am using both the app directory and the page directory. However, I am encountering an issue with SPA navigation between the app and page directories. Here are the details:

**Project Overview:**

Utilizing both the app directory and the page directory.
Using the new App Router in the app directory and the traditional Page Router in the page directory.

**Issue Details:**

When transitioning from a page within the app directory to a page within the page directory, or vice versa, the SPA navigation does not work as expected, and the page fully reloads.
I have attempted to use next/link and next/router for the transitions.
If anyone has experienced this issue or knows how to achieve smoother SPA transitions between the app and page directories, I would greatly appreciate your assistance!

**Note:**

Even after setting clientRouterFilter to true as per the Next.js documentation on "Routing Between Pages and App," the issue with SPA transitions between the app directory and page directory remains unresolved.
https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#7-routing-between-pages-and-app

**Comment by icyJoseph** [comment]
Hi,

This is, somewhat, expected behavior.

The App Router and Pages Router, are fundamentally different router implementations, and use different React versions, AFAIK, and navigating between them, requires page unloading.

**Reply by shunshimono** [reply]
Thank you for your response. However, this means that a gradual migration between the App Router and Pages Router is not possible😢

**Reply by icyJoseph** [reply]
How so? what is particularly damaging for your use case? How do you interpret gradual migration within your product?

**Reply by shunshimono** [reply]
The challenge with gradual migration in my use case is that transitioning between the App Router and Pages Router results in a full page reload, which impacts the user experience and state management. This disruption affects the seamless SPA experience and the continuity of application state.

For my product, gradual migration means incrementally updating parts of the application to use the new App Router while maintaining the old Pages Router in other parts.
